### PR TITLE
Add privileges for Fleet example package for transforms

### PR DESCRIPTION
**Pull request by droberts195**
As a companion to elastic/kibana#140046 an example transform package is being created in elastic/integrations#4138.

Initially the intention was that this example would use the Kibana sample data. Then, after realising that wouldn't work due to lack of permissions, it was changed to reinstall a transform that is created by the security solution. However, it has been pointed out that this is dangerous because even if we tell people not to install the example in production it could somehow happen by mistake and then we'd have created objects that clash with production code.

Therefore, it is best that the example transform package be switched back to the original idea of using the Kibana sample data, so that there is no risk of it ever clashing with a production package.

### CODE REVIEW

**Comment by droberts195** [comment]
cc @qn895 @peteharverson @sophiec20

**Comment by slobodanadamovic** [comment]
Pinging @elastic/kibana-security  for awareness.

**Review by slobodanadamovic — APPROVED** [review]
LGTM, but we should still wait for approval from @elastic/kibana-security team as this change affects `kibana_system` user.

**Review by azasypkin — APPROVED** [review]
LGTM, thanks for the ping @slobodanadamovic!

**Pull request merged**
